### PR TITLE
Fully support `||` and `&&` in `pluginToggleBooleanFlag`

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This is mostly needed this to cleanup https://github.com/babel/babel/pull/15955. Review with whitespace diff disabled.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15961"></a>

### CODE REVIEW

File: packages/babel-preset-env/src/available-plugins.ts [hunk 1]
```diff
@@ -152,59 +150,74 @@ if (!process.env.BABEL_8_BREAKING) {
...
+        : () => require("@babel/plugin-syntax-top-level-await"),
   });
 
   // This is a CJS plugin that depends on a package from the monorepo, so it
   // breaks using ESM. Given that ESM builds are new enough to have this
   // syntax enabled by default, we can safely skip enabling it.
   if (!USE_ESM) {
     // @ts-expect-error unknown key
-    availablePlugins["unicode-sets-regex"] = USE_ESM_OR_STANDALONE
+    availablePlugins["unicode-sets-regex"] = IS_STANDALONE
```
Comment (liuxingbaoyu) [inline packages/babel-preset-env/src/available-plugins.ts:220]: Was this change intentional?

**Review by liuxingbaoyu — APPROVED** [review]
Amazing!

**Review by JLHwung — APPROVED** [review]
